### ISSUE TITLE
[Feature request] Configurable route / action completion for custom methods

### ISSUE THREAD

**Issue by stayallive** [laravel-idea/plugin#208]
Hey! We are using your plugin in our team and are loving it :)

We have a maybe over specific feature request but I do want to drop it here in case you find it worth implementing because I think we are not the only ones with this use case.

We have a few custom helpers methods defined like:

```php
function actionForTenant($action, ?Tenant $tenant = null, $parameters = [], $absolute = true);
function routeForTenant($route, ?Tenant $tenant = null, $parameters = [], $absolute = true);
```

These are exactly the same as `action()` and `route()` where the first argument could use the autocompletion and reference searches. The next arguments are a bit different but I don't think there is any autocompletion happening there so I would assume that won't matter much.

Long story short, I would like to request a configurable list of `action()` and `route()` like methods that would really help us out. If you have a better idea on how to achieve this I'm of course also all ears!

I hope this is something you would consider adding in a future version 👍

**Comment by adelf** [comment]
Hi. I'm thinking about it a lot. Configurable completions for routes, views, configs, etc. 
I hope to implement something about it soon. Thanks!

**Comment by stayallive** [comment]
I hadn't even considered views and configs, but yeah that would also be wonderful!

**Comment by adelf** [comment]
With the new version (4.1.1 released today) you can just add an `ide.json` file somewhere in the project with this content:

```
{
    "$schema": "http://laravel-ide.com/schema/laravel-ide-v1.json",
    "completion": {
        "route": [
            {
                "methods": ["routeForTenant"],
                "parameters": [1]
            }
        ],
        "routeAction": [
            {
                "methods": ["actionForTenant"],
                "parameters": [1]
            }
        ]
    }
}
```

$schema parameter is not required but it will add an autocompletion for all objects. This file will add route completion for the first parameter for *all* `routeForTenant` methods. The same is for routeAction. This is an experimental feature. Full documentation will be available soon.

**Comment by stayallive** [comment]
This is crazy good! I love it! Works like a charm 😍 

Please let me know once the docs are up, we're going to make good use out of this 💪

**Comment by adelf** [comment]
@stayallive please check - https://laravel-idea.com/docs/ide_json

**Comment by stayallive** [comment]
This is amazing, thanks a bunch! Very happy with this flexibility and power we have with this!
